code/modules/codex/entries/machinery.dm

```diff
@@ -0,0 +1,127 @@
+/obj/machinery/get_antag_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.antag_text)
+		return general_entry.antag_text
+
+/obj/machinery/get_lore_info()
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.lore_text)
+		return general_entry.lore_text
+
+/obj/machinery/get_mechanics_info()
+	var/list/machinery_strings = list()
+
+	var/list/entries = SScodex.retrieve_entries_for_string(name)
+	var/datum/codex_entry/general_entry = LAZYACCESS(entries, 1)
+	if(general_entry && general_entry.mechanics_text)
+		machinery_strings += general_entry.mechanics_text + "<br>"
+
+	if(use_power)
+		machinery_strings += "This requires power to function."
+
+	if(idle_power_usage)
+		machinery_strings += "It uses [idle_power_usage] power when idle and [active_power_usage] power when active."
+
+	if(machine_max_charge)
+		machinery_strings += "It can hold [machine_max_charge] power in it's internal buffer."
+
+	if(destructible)
+		machinery_strings += "This can be destroyed."
+
+	if(wrenchable == TRUE)
+		machinery_strings += "This can be wrenched to allow it to be moved."
+	else
+		machinery_strings += "This cannot be wrenched to allow it to be moved."
+/*
+	if(length(req_access) > 0)
+		machinery_strings += "<br><U>You have to have one of the following access requirements to access this normally</U>:"
+		is_in_access_list(req_access)
+
+	if(length(req_one_access) > 0)
+		machinery_strings += "<br><U>You have to have one of the following access requirements to access this normally</U>:"
+		is_in_access_list(req_one_access)
+*/
+	if(length(component_parts) > 0)
+		machinery_strings += "<U>It is made from the following parts</U>:"
+		for(var/X in component_parts)
+			var/obj/A = X
+			machinery_strings += "[initial(text2path(A).name)]"
+
+	. += jointext(machinery_strings, "<br>")
+
+/datum/codex_entry/emitter
+	associated_paths = list(/obj/machinery/power/emitter)
+	mechanics_text = "You must secure this in place with a wrench and weld it to the floor before using it. The emitter will only fire if it is installed above a cable endpoint. Clicking will toggle it on and off, at which point, so long as it remains powered, it will fire in a single direction in bursts of four."
+	lore_text = "Lasers like this one have been in use for ages, in applications such as mining, cutting, and in the startup sequence of many advanced space station and starship engines."
+	antag_text = "This baby is capable of slicing through walls, sealed lockers, and people."
+
+/* do not use until someone reworks access. Then do something better.
+proc/is_in_access_list(var/list/X)
```

Comment:
Need help with this?

Reply:
Spooky told me to not use it until access was reworked to be better.